src/flow-worker.c

```diff
@@ -430,6 +430,10 @@ static inline void FlowWorkerStreamTCPUpdate(ThreadVars *tv, FlowWorkerThreadDat
             TmqhOutputPacketpool(tv, x);
         }
     }
+    if (FlowChangeProto(p->flow) && p->flow->flags & FLOW_ACTION_DROP) {
+        // in case f->flags & FLOW_ACTION_DROP was set by one of the dequeued packets
+        PacketDrop(p, ACTION_DROP, PKT_DROP_REASON_FLOW_DROP);
```

Comment:
this drop reason isn't correct, it should pass on  the original reason given by the tunnel packet

Reply:
What do you call a "tunnel" packet ? There is no GRE or over tunneling protocols here...

Why is not the reason correct ?
`FlowHandlePacketUpdate` does
```
    if (f->flags & FLOW_ACTION_DROP) {
        PacketDrop(p, ACTION_DROP, PKT_DROP_REASON_FLOW_DROP);
    }
```

same reason for same condition : flow flags have `FLOW_ACTION_DROP` 

so I do not put reason `PKT_DROP_REASON_RULES`